android/src/main/kotlin/com/kasem/media_picker_builder/providers/ThumbnailImageProvider.kt

```diff
@@ -59,123 +62,101 @@ object ThumbnailImageProvider {
-                                + " AND ${MediaStore.Images.Thumbnails.KIND} = ${MediaStore.Images.Thumbnails.MINI_KIND}",
-                        null,
-                        null)?.use { cursor ->
-                    if (cursor.count > 0) {
-                        cursor.moveToFirst()
-                        path = cursor.getString(0)
-                    }
-                }
-            }
-            MediaFile.MediaType.VIDEO -> {
-                context.contentResolver.query(
-                        MediaStore.Video.Thumbnails.EXTERNAL_CONTENT_URI,
-                        arrayOf(MediaStore.Video.Thumbnails.DATA),
-                        "${MediaStore.Video.Thumbnails.VIDEO_ID} = $fileId AND "
-                                + "${MediaStore.Video.Thumbnails.KIND} = ${MediaStore.Video.Thumbnails.MINI_KIND}",
-                        null,
-                        null)?.use { cursor ->
-                    if (cursor.count > 0) {
-                        cursor.moveToFirst()
-                        path = cursor.getString(0)
-                    }
-                }
-            }
+        val dir = File(context.externalCacheDir, ".thumbnails")
+        if (!dir.exists()) {
+            dir.mkdirs()
+        }
+        val outputFile = File(dir, "$fileId.jpg")
+        if (outputFile.exists()) {
+            return outputFile.path
+        } else {
+            return generateThumbnail(context, fileId, type, outputFile)
         }
-        return path
     }
 
-    @Throws(Exception::class)
-    private fun generateThumbnail(context: Context, fileId: Long, type: MediaFile.MediaType): String? {
+    private fun generateThumbnail(
+            context: Context,
+            fileId: Long,
+            type: MediaFile.MediaType,
+            outputFile: File
+    ): String? {
+        val bitmap: Bitmap
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            val dir = File(context.externalCacheDir, ".thumbnails")
-            if (!dir.exists()) {
-                dir.mkdirs()
-            }
-            val outputFile = File(dir, "$fileId.jpg")
-            if (outputFile.exists()) return outputFile.path
-
-            // Generate thumbnail
-            val bitmap = when (type) {
+            bitmap = when (type) {
                 MediaFile.MediaType.IMAGE -> {
                     val uri = Uri.parse("${MediaStore.Images.Media.EXTERNAL_CONTENT_URI}/$fileId")
-                    context.contentResolver.loadThumbnail(uri, Size(90, 90), null) // TODO: handle cancelling
+                    context.contentResolver.loadThumbnail(uri, Size(90, 90), null)
                 }
                 MediaFile.MediaType.VIDEO -> {
                     val uri = Uri.parse("${MediaStore.Video.Media.EXTERNAL_CONTENT_URI}/$fileId")
-                    context.contentResolver.loadThumbnail(uri, Size(270, 270), null) // TODO: handle cancelling
+                    context.contentResolver.loadThumbnail(uri, Size(270, 270), null)
```

Comment:
Why remove the comment?

Reply:
1. This todo doesn't provide much.
2. Cancelling is not needed as it's quick operation
3. Caneclling would be very hard to do (flutter-android communication)